portal-web/test/functional/com/liferay/portalweb/macros/SAML.macro

```diff
@@ -1,5 +1,132 @@
 definition {
 
+	@description = "Add Identity Provider Configurations to Service Provider"
+	macro addIDPConfigurationsToServiceProvider {
+		if (isSet(userEmailAddress)) {
+			User.firstLoginUI(
+				password = "test",
+				specificURL = "${spURL}",
+				userEmailAddress = "${userEmailAddress}");
+		}
+		else {
+			User.firstLoginUI(
+				password = "test",
+				specificURL = "${spURL}",
+				userEmailAddress = "test@liferay.com");
+		}
+
+		Navigator.openSpecificURL(url = "${spURL}");
+
+		ApplicationsMenu.gotoPortlet(
+			category = "Security",
+			panel = "Control Panel",
+			portlet = "SAML Admin");
+
+		SAMLNavigation.gotoIdentityProviderConnections();
+
+		SAMLPortlet.addIdentityProviderConnection(
+			idpAttributeMapping = "${idpAttributeMapping}",
+			idpEntityID = "${idpEntityID}",
+			idpKeepAliveURL = "${idpKeepAliveURL}",
+			idpMetadataURL = "${idpURL}/c/portal/saml/metadata",
+			idpNameID = "${idpEntityID}",
+			index = "${index}",
```

Comment:
The name index could be more specific.

Reply:
Renamed to: samlAttributeIDforUserMatching